framework/src/meshgenerators/BlockToMeshConverterGenerator.C

```diff
@@ -0,0 +1,125 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BlockToMeshConverterGenerator.h"
+#include "CastUniquePointer.h"
+#include "libmesh/elem.h"
+#include "MooseMeshUtils.h"
+
+registerMooseObject("MooseApp", BlockToMeshConverterGenerator);
+
+InputParameters
+BlockToMeshConverterGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addClassDescription(
+      "Converts a block (subdomain) from a mesh into a stand-alone mesh with one block in it.");
+
+  // list params
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<std::vector<SubdomainName>>(
+      "target_blocks",
+      "The (list of) blocks (or 'subdomains') we wish to have moved to a new mesh (by name, not "
+      "ID)");
+
+  return params;
+}
+
+BlockToMeshConverterGenerator::BlockToMeshConverterGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _target_blocks(getParam<std::vector<SubdomainName>>("target_blocks"))
+{
+}
+
+std::unique_ptr<MeshBase>
+BlockToMeshConverterGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+
+  if (!mesh->is_replicated())
+    mooseError("BlockToMeshConverterGenerator is not implemented for distributed meshes.");
```

Comment:
Would it be sufficient to just throw on a `MeshSerializer` here?  "Doesn't work efficiently" is still a step up from "doesn't work at all".

Reply:
Well, there's support for distributed meshes, and then there's "support" for distributed meshes.  Temporarily serializing a mesh is an awful excuse for support ... but it's better than nothing, especially since proper distributed mesh support can be a *lot* of work.  In the context of a mesh generator temporary serialization often makes sense, when users have a workflow where they construct a small (and hence cheap to serialize) mesh at one stage, and only refine or extrude or stitch or whatever to turn it into a large (and hence necessary to distribute) mesh at a later stage.

Reply:
This is a good idea.

We likely should remove all these blanket mooseError statements and add serializers in all the MGs instead as a first step towards having fully distributed generation.